### PR TITLE
GET /api/articles

### PR DESCRIPTION
**Pull request by muxappo**
Was a little stuck on what errors could be encountered with this one, if you can give any hints besides the 404 I tested by temp deleting the data within the test that'd be great

### CODE REVIEW

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -123,4 +123,62 @@ describe("GET requests", () => {
...
+      DELETE FROM comments;
+      DELETE FROM articles;`
+        )
+        .then(() => {
+          return request(app)
+            .get("/api/articles")
+            .expect(404)
+            .then(({ body: { msg } }) => {
+              expect(msg).toBe("No articles found");
+            });
```
Comment (Harrison365) [inline __tests__/integration.test.js]: you don't need to do this :) 
There are no sad paths possible besides the 404 you've already looked at :)

File: models/models.js [hunk 2]
```diff
@@ -22,7 +22,38 @@ exports.fetchArticleByID = (id) => {
...
+    articles.created_at,
+    articles.votes,
+    articles.article_img_url,
+  COUNT(comments.comment_id) AS comment_count
+  FROM articles
+  LEFT JOIN comments ON articles.article_id = comments.article_id
+  GROUP BY articles.article_id
+  ORDER BY created_at DESC
+  `;
+
```
Comment (Harrison365) [inline models/models.js:50]: Nice SQL 👍

File: models/models.js [hunk 3]
```diff
@@ -22,7 +22,38 @@ exports.fetchArticleByID = (id) => {
...
+  return db.query(getQuery).then(({ rows }) => {
+    if (rows.length === 0) {
+      return Promise.reject({
+        status: 404,
+        message: "No articles found",
+      });
     }
```
Comment (Harrison365) [inline models/models.js]: Shouldn't be necessary :)

File: __tests__/integration.test.js [hunk 4]
```diff
@@ -123,4 +123,62 @@ describe("GET requests", () => {
...
+        });
+    });
+
+    it("returns articles in descending order by date", () => {
+      return request(app)
+        .get("/api/articles")
+        .then(({ body: { articles } }) => {
+          expect(articles).toBeSortedBy("created_at", { descending: true });
+        });
+    });
```
Comment (Harrison365) [inline __tests__/integration.test.js:176]: Nice 👍

File: __tests__/integration.test.js [hunk 5]
```diff
@@ -123,4 +123,62 @@ describe("GET requests", () => {
...
+
+    it("returned objects have a call_count property", () => {
+      return request(app)
+        .get("/api/articles")
+        .then(({ body: { articles } }) => {
+          for (let i = 0; i < articles.length; i++) {
+            expect(articles[i].hasOwnProperty("comment_count")).toBe(true);
+          }
+        });
```
Comment (Harrison365) [inline __tests__/integration.test.js:167]: We can tidy this I think. 

Try using a forEach then checking the articles using...
expect.objectContaining({
    article_id: expect.any(Number),
    etc...

**Comment by Harrison365** [comment]
Nice work 😄 
Just a few things to tidy 👍

File: __tests__/integration.test.js (hunk 1 above)
Comment (muxappo) [reply __tests__/integration.test.js]: I did think this, but the task specifically mentioned considering errors so I got a bit stuck thinking there should be something. It did feel extraneous once I'd tested for it though

**Review by Beth-McCormack — APPROVED** [review]

**Pull request merged**